src/plugins/vis_type_timeseries/public/application/components/annotation_row.tsx

```diff
@@ -0,0 +1,278 @@
+import { getDefaultQueryLanguage } from './lib/get_default_query_language';
+
+// @ts-expect-error not typed yet
+import { IconSelect } from './icon_select/icon_select';
+
+import type { Annotation, FetchedIndexPattern, IndexPatternValue } from '../../../common/types';
+import type { VisFields } from '../lib/fetch_fields';
+
+const RESTRICT_FIELDS = [KBN_FIELD_TYPES.DATE];
+const INDEX_PATTERN_KEY = 'index_pattern';
+
+export interface AnnotationRowProps {
+  annotation: Annotation;
+  fields: VisFields;
+  onChange: (partialModel: Partial<Annotation>) => void;
+  handleAdd: () => void;
+  handleDelete: () => void;
+}
+
+const getAnnotationDefaults = () => ({
+  fields: '',
+  template: '',
+  index_pattern: '',
+  query_string: { query: '', language: getDefaultQueryLanguage() },
+});
+
+export const AnnotationRow = ({
+  annotation,
+  fields,
+  onChange,
+  handleAdd,
+  handleDelete,
+}: AnnotationRowProps) => {
+  const model = useMemo(() => ({ ...getAnnotationDefaults(), ...annotation }), [annotation]);
+  const htmlId = htmlIdGenerator(model.id);
+
+  const [fetchedIndex, setFetchedIndex] = useState<FetchedIndexPattern | null>(null);
+
+  useEffect(() => {
+    const updateFetchedIndex = async (index: IndexPatternValue) => {
+      const { indexPatterns } = getDataStart();
+
+      setFetchedIndex(
+        index
+          ? await fetchIndexPattern(index, indexPatterns)
+          : {
+              indexPattern: undefined,
+              indexPatternString: undefined,
+            }
+      );
+    };
+
+    updateFetchedIndex(model.index_pattern);
+  }, [model.index_pattern]);
+
+  const togglePanelActivation = useCallback(
+    () =>
+      onChange({
+        hidden: !model.hidden,
+      }),
+    [model.hidden, onChange]
+  );
+
+  const handleChange = useCallback(
+    (name: string) => (
+      event: Array<EuiComboBoxOptionOption<string>> | ChangeEvent<HTMLInputElement>
+    ) =>
+      onChange({
+        [name]: Array.isArray(event) ? event?.[0]?.value : event.target.value,
+      }),
+    [onChange]
+  );
+
+  const handleQueryChange = useCallback(
+    (filter: Query) =>
+      onChange({
+        query_string: filter,
+      }),
+    [onChange]
+  );
+
+  return (
+    <div className="tvbAnnotationsEditor" key={model.id}>
+      <EuiFlexGroup responsive={false}>
+        <EuiFlexItem grow={false}>
+          <ColorPicker disableTrash={true} onChange={onChange} name="color" value={model.color} />
+        </EuiFlexItem>
+
+        <EuiFlexItem className="tvbAggRow__children">
+          <EuiFlexGroup responsive={false} wrap={true} gutterSize="m">
+            <EuiFlexItem>
+              <IndexPatternSelect
+                value={model[INDEX_PATTERN_KEY]}
+                indexPatternName={INDEX_PATTERN_KEY}
+                onChange={onChange}
+                fetchedIndex={fetchedIndex}
+              />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <FieldSelect
+                type={'time_field'}
```

Comment:
nit: No need to wrap it, type='time_field' and as we have done it for ```const INDEX_PATTERN_KEY = 'index_pattern';``` we could do the same here